### ISSUE TITLE
[libc++][test] std/containers/views/mdspan/mdspan/conversion.pass.cpp Fails on PowerPC targets

### ISSUE THREAD

**Issue by amy-kwan** [llvm/llvm-project#64427]
This test was introduced in https://reviews.llvm.org/D154367.

Assertion seen:
```
Command Output (stdout):
--
$ ":" "COMPILED WITH"
$ "/home/amyk/llvm/community/build/bootstrap/stage1/build/./bin/clang++" "/home/amyk/llvm/community/llvm-project/libcxx/test/std/containers/views/mdspan/mdspan/conversion.pass.cpp" "-pthread" "--target=powerpc64le-unknown-linux-gnu" "-nostdinc++" "-I" "/home/amyk/llvm/community/build/bootstrap/stage1/build/include/c++/v1" "-I" "/home/amyk/llvm/community/build/bootstrap/stage1/build/include/powerpc64le-unknown-linux-gnu/c++/v1" "-I" "/home/amyk/llvm/community/llvm-project/libcxx/test/support" "-std=c++26" "-Werror" "-Wall" "-Wctad-maybe-unsupported" "-Wextra" "-Wshadow" "-Wundef" "-Wunused-template" "-Wno-unused-command-line-argument" "-Wno-attributes" "-Wno-pessimizing-move" "-Wno-noexcept-type" "-Wno-atomic-alignment" "-Wno-reserved-module-identifier" "-Wno-user-defined-literals" "-Wno-tautological-compare" "-Wsign-compare" "-Wunused-variable" "-Wunused-parameter" "-Wunreachable-code" "-Wno-unused-local-typedef" "-Wno-local-type-template-args" "-Wno-c++11-extensions" "-Wno-unknown-pragmas" "-D_LIBCPP_HAS_NO_PRAGMA_SYSTEM_HEADER" "-D_LIBCPP_ENABLE_EXPERIMENTAL" "-Werror=thread-safety" "-Wuser-defined-warnings" "-lc++experimental" "-nostdlib++" "-L" "/home/amyk/llvm/community/build/bootstrap/stage1/build/./lib/powerpc64le-unknown-linux-gnu" "-Wl,-rpath,/home/amyk/llvm/community/build/bootstrap/stage1/build/./lib/powerpc64le-unknown-linux-gnu" "-lc++" "-o" "/home/amyk/llvm/community/build/bootstrap/stage1/build/runtimes/runtimes-bins/test/std/containers/views/mdspan/mdspan/Output/conversion.pass.cpp.dir/t.tmp.exe"
$ ":" "EXECUTED AS"
$ "/usr/bin/python3.8" "/home/amyk/llvm/community/llvm-project/libcxx/test/../utils/run.py" "--execdir" "/home/amyk/llvm/community/build/bootstrap/stage1/build/runtimes/runtimes-bins/test/std/containers/views/mdspan/mdspan/Output/conversion.pass.cpp.dir" "--" "/home/amyk/llvm/community/build/bootstrap/stage1/build/runtimes/runtimes-bins/test/std/containers/views/mdspan/mdspan/Output/conversion.pass.cpp.dir/t.tmp.exe"
# command stderr:
t.tmp.exe: /home/amyk/llvm/community/llvm-project/libcxx/test/std/containers/views/mdspan/mdspan/conversion.pass.cpp:87: void test_conversion(FromMDS) [ToMDS = std::mdspan<const int, std::extents<int>, std::layout_left, conv_test_accessor_c<int, false, true, true, true>>, FromMDS = std::mdspan<int, std::extents<int>, std::layout_left, conv_test_accessor_nc<int, true, true>>]: Assertion `to_mds.data_handle() == from_mds.data_handle()' failed.

error: command failed with exit status: 250
```

This issue occurs on both LLVM 17.0.0 rc1 and `main`.

**Comment by amy-kwan** [comment]
@crtrott Just wanted to let you know that I have discovered this when testing libc++.

**Comment by crtrott** [comment]
This looks suspiciously simmilar to the MinGW issue reported by me. I will have a look.

**Comment by crtrott** [comment]
We should try the godbolt example seen in #64077 on PowerPC and check whether it has the same issue there.

**Comment by crtrott** [comment]
I confirmed that the godbolt example https://godbolt.org/z/KK8aj5bs7 (minus the concept and span include which isn't needed) fails on PowerPC with Clang 14 too. No extra options needed just `clang++ main.cpp; ./a.out` will do it.

**Comment by mstorsjo** [comment]
https://reviews.llvm.org/D157332 has got a potential fix for this issue.

**Comment by amy-kwan** [comment]
I can confirm that https://reviews.llvm.org/D157332 resolves the issue.

**Comment by tru** [comment]
Can this be closed now? I see linked commits to this issue.

**Comment by mstorsjo** [comment]
> Can this be closed now? I see linked commits to this issue.

Yes, some version of the fix was confirmed as fixing this issue, and as the final forms of the fix, in 4d502f14aca55b8378e75373639cbee305c8c2ad, did essentially the same thing, I think we can close it, and reopen if it (unlikely) turns out to not be fixed after all.
